app/services/user_event_creator.rb

```diff
@@ -76,9 +76,7 @@ def build_disavowal_token
 
   # @return [Array(Event, String)] an (event, disavowal_token) tuple
   def create_event_for_new_device(event_type:, user:, disavowal_token:)
-    user_has_multiple_devices = user.recent_devices?
-
-    if user_has_multiple_devices && disavowal_token.nil?
+    if user.fully_registered? && user.recent_devices? && disavowal_token.nil?
```

Comment:
I liked pulling `user.recent_devices?` into a named variable for clarity, I think we could event make it clearer if we said something like
```suggestion
    has_existing_devices = user.recent_devices?
    if user.fully_registered? && has_existing_devices && disavowal_token.nil?
```
or since I think this is the only use of `User#recent_devices?` we could just rename that method to something like `User#has_existing_devices?` for readability here?

Reply:
Yeah, I was gonna say, the fact that we felt the need to assign it to a variable of a different name than the method itself seems like a smell on the method name. I'd probably move that direction to rename the method itself.

Reply:
Updated to `has_devices?` in 61ff910.